src/proc/mod.rs

```diff
@@ -108,6 +118,37 @@ impl super::TypeInner {
             Self::Image { .. } | Self::Sampler { .. } => 0,
         }
     }
+
+    /// Return the length of a subscriptable type, if known at compile time.
+    ///
+    /// The value returned is appropriate for bounds checks on subscripting. If
+    /// the length is dynamic (say, a WGSL runtime-sized array), then return
+    /// `None`.
+    ///
+    /// `self` must be a vector, matrix, or array.
+    pub fn to_known_length(&self, module: &super::Module) -> Option<u32> {
```

Comment:
probably needs a rename to something like `array_access_length` to indicate that we are talking about array-like types here, specifically. Otherwise, it's not clear (judging by the name) why structures are excluded, for example

Reply:
Makes sense. Renamed to `array_known_length`. I think `known` is an important component of the name.